pkg/reporter/reporter.go

```diff
@@ -0,0 +1,106 @@
+package reporter
+
+import (
+	"github.com/codefresh-io/cli-v2/pkg/log"
+	"github.com/codefresh-io/cli-v2/pkg/store"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+var ar AnalyticsReporter
+
+type (
+	AnalyticsReporter interface {
+		ReportStep(CliStepData)
+	}
+
+	CliStepData struct {
+		Event       CliEventType
+		Status      CliStepStatus
+		Description string
+		Err         error
+	}
+
+	segmentAnalyticsReporter struct {
+		client    analytics.Client
+		userId    string
+		accountId string
+	}
+
+	noopAnalyticsReporter struct{}
+
+	CliEventType  string
+	CliStepStatus string
+)
+
+const (
+	// Install
+	PRE_INSTALLATION_CHECKS        CliEventType = "install.pre-installation-checks"
```

Comment:
Please follow the golang coding convention when naming consts:
https://stackoverflow.com/questions/22688906/go-naming-conventions-for-const

Reply:
Also, think of having prefix inside the name to indicate if it's an install or uninstall step, because they have the same type it might be confusing if there are similar steps.
For example: `InstallStepPreChecks`, `InstallStepDownloadRuntimeDefinition`
And: `UninstallStepCheckRuntimeExists`....